app/renderer/lib/client-frameworks/robot.js

```diff
@@ -50,285 +43,314 @@ ${this.indent(code, 4)}
+      '-ios predicate string': 'nsp',
+      '-ios class chain': 'chain',
     };
     if (!suffixMap[strategy]) {
       return this.handleUnsupportedLocatorStrategy(strategy, locator);
     }
-    //TODO: in the robot case, we need the ID on the codeFor_ for execution
-    this.lastID = `${strategy}=${locator}`;
-
-    if (this.lastID.includes('accessibility id')) {
-      this.lastID = this.lastID.replace('accessibility id', 'accessibility_id');
-    }
 
-    return `# ${this.lastID}`;
+    return `$\{${localVar}} =    Set Variable     ${suffixMap[strategy]}=${locator}`;
   }
 
-  getApplicationInitialization () {
-    let varString = Object.keys(this.caps).map((k) => `${k}=\$\{${k}\}`).join('  ');
-    return `    Open Application    \$\{REMOTE_URL\}   ${varString}`;
+  codeFor_click (varName, varIndex) {
+    return `Click Element    $\{${this.getVarName(varName, varIndex)}}`;
   }
 
-  codeFor_executeScript (/*varNameIgnore, varIndexIgnore, args*/) {
-    return `    Execute Script    TODO implement executeScript`;
-  }
-
-
-  codeFor_click (/*varName, varIndex*/) {
-    return `    Click Element    ${this.lastID}`;
-  }
-
-  codeFor_clear (/*varName, varIndex*/) {
-    return `    Clear Text    ${this.lastID}`;
+  codeFor_clear (varName, varIndex) {
+    return `Clear Text    $\{${this.getVarName(varName, varIndex)}}`;
   }
 
   codeFor_sendKeys (varName, varIndex, text) {
-    return `    Input Text    ${this.lastID}    ${text}`;
-  }
-
-  codeFor_back () {
-    return `    Go Back`;
+    return `Input Text    $\{${this.getVarName(varName, varIndex)}}    ${text}`;
   }
 
   codeFor_tap (varNameIgnore, varIndexIgnore, pointerActions) {
     const {x, y} = this.getTapCoordinatesFromPointerActions(pointerActions);
-
-    return `    Tap With Positions    100    \$\{${x}, ${y}\}`;
+    return `@{finger} =    Create List    $\{${x}}    $\{${y}}
+@{positions} =    Create List    $\{finger}
+Tap With Positions    $\{100}    $\{positions}`;
   }
 
   codeFor_swipe (varNameIgnore, varIndexIgnore, pointerActions) {
     const {x1, y1, x2, y2} = this.getSwipeCoordinatesFromPointerActions(pointerActions);
+    return `Swipe    $\{${x1}}    $\{${y1}}    $\{${x2}}    $\{${y2}}`;
+  }
+
+  // Execute Script
+
+  codeFor_executeScriptNoArgs (scriptCmd) {
+    return `Execute Script    ${scriptCmd}`;
+  }
 
-    return `    Swipe    ${x1}    ${y1}    ${x2}    ${y2}`;
+  codeFor_executeScriptWithArgs (scriptCmd, jsonArg) {
+    // change the JSON object into a format accepted by Create Dictionary: a sequence of key=value
+    const argsValuesStrings = _.map(jsonArg[0], (v, k) => `${k}=${this.getRobotVal(v)}`);
+    return `&{scriptArgument}    Create Dictionary    ${argsValuesStrings.join('    ')}
+Execute Script    ${scriptCmd}    &{scriptArgument}`;
   }
 
-  // TODO: Add these robot framework commands
+  // App Management
+
   codeFor_getCurrentActivity () {
-    return '';
+    return '${activity_name} =    Get Activity';
```

Comment:
I assume we can still call `mobie: ` extensions from RF

Reply:
same below

Reply:
I think we can, but I'm not really sure of the best approach. The user might have an older driver version that does not support all of the replacement extensions. Ideally I would like to have separate GUI buttons for the various `mobile` extension commands, and leave the current buttons linked to their original, deprecated functionality, with added deprecation labels.
(I also realise I already replaced many such deprecated methods in #1173, so now I'm not sure if I should change that back)

Reply:
if they have older driver then an error message would be shown, so this is ok. Lets always aim the most recent version available

Reply:
I've now changed all the methods for which the equivalent Ruby code was calling mobile extensions. Common methods (i.e. without language-specific variable assignment) were moved to the common class file.